shopify/slack-notification-template/api/models/shopifyShop/actions/setSlackChannel.ts

```diff
@@ -3,49 +3,39 @@ import {
   preventCrossShopDataAccess,
   save,
   ActionOptions,
-  SetSlackChannelShopifyShopActionContext,
 } from "gadget-server";
 import { slackClient } from "../../../../utilities";
 
-/**
- * @param { SetSlackChannelShopifyShopActionContext } context
- */
-export async function run({ params, record, logger, api }) {
+export const run: ActionRun = async ({ params, record, logger, api }) => {
   applyParams(params, record);
   await preventCrossShopDataAccess(params, record);
 
   if (record.changed("slackChannelId")) {
-    const { previous } = record.changes("slackChannelId");
+    const changes = record.changes("slackChannelId");
+    const previous = changes.changed ? changes.previous : null;
+
     try {
       if (previous) {
         await slackClient.conversations.leave({
-          token: record.slackAccessToken,
+          token: record.slackAccessToken as string,
```

Comment:
A better way to handle this might be to perform a check earlier and throw an error if the slackAccessToken is null or undefined

Reply:
I changed the token to check in the if